flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/ExecutionSlotAllocationContext.java

```diff
@@ -44,9 +45,9 @@
      * Returns prior allocation id for an execution vertex.
      *
      * @param executionVertexId id of the execution vertex
-     * @return prior allocation id for the given execution vertex
+     * @return prior allocation id for the given execution vertex if it exists
      */
-    AllocationID getPriorAllocationId(ExecutionVertexID executionVertexId);
+    Optional<AllocationID> getPriorAllocationId(ExecutionVertexID executionVertexId);
```

Comment:
OT: just one thought on naming, I always tried to stick with `find...` prefix for anything that can return optional, to make it more explicit that you may not find the result 🤔 

But this is not really a common pattern in Flink, so feel free to ignore it.

Reply:
Sounds good to me. Will change the method names accordingly.